encord/objects/common.py

```diff
@@ -196,8 +191,9 @@ class RadioAttribute(Attribute):
 
     options: List[NestableOption] = field(default_factory=list)
 
-    def get_property_type(self) -> PropertyType:
-        return PropertyType.RADIO
+    @staticmethod
+    def get_property_type_name() -> str:
+        return "radio"
```

Comment:
This was better before?

Reply:
Don't think this was better, as you had to update two separate places, hence it is easier to get out of sync and put a bug.
So I'd prefer to keep it all in one place: you have a subclass to define a certain property type, and it contains all necessary information. Rather than spread this all over the codebase

Reply:
But good point, made it private to indicate this is not used anywhere outside of the class hierarchy

Reply:
I would usually go with an enum, and refer to the enum throughout the codebase, which then makes it very easy to spot all the places where we're doing something with a PropertyType and tracing it across multiple functions. Using string somewhat loses that since we need to do string searches, and union of string searches (if we're looking for all the places a PropertyType would get used). Particularly when they get passed around as "str" type arguments in functions or objects, the trail gets lost a bit. 

In this case I don't think it has that much of an impact since the scope is quite small, I'm actually fine with the change.

Reply:
I would usually go with an enum, and refer to the enum throughout the codebase, which then makes it very easy to spot all the places where we're doing something with a PropertyType and tracing it across multiple functions. Using string somewhat loses that since we need to do string searches, and union of string searches (if we're looking for all the places a PropertyType would get used). Particularly when they get passed around as "str" type arguments in functions or objects, the trail gets lost a bit. 

In this case I don't think it has that much of an impact since the scope is quite small, I'm actually fine with the change.

edit1: and having it in one place is a small benefit so it's not like we're not getting anything in return

Reply:
I would usually go with an enum, and refer to the enum throughout the codebase, which then makes it very easy to spot all the places where we're doing something with a PropertyType and tracing it across multiple functions. Using string somewhat loses that since we need to do string searches, and union of string searches (if we're looking for all the places a PropertyType would get used). Particularly when they get passed around as "str" type arguments in functions or objects, the trail gets lost a bit. 

In this case I don't think it has that much of an impact since the scope is quite small, I'm actually fine with the change.

edit1: and having it in one place is a benefit so it's not like we're not getting anything in return